File: python/agentshield/policy.py
from dataclasses import dataclass, field
from typing import Literal, Optional


@dataclass
class Policy:
    tool_allowlist: Optional[list[str]] = None
    tool_denylist: list[str] = field(default_factory=list)
    memory_ttl: int = 3600  # seconds; 0 = no expiry
    injection_sensitivity: Literal["low", "medium", "high"] = "medium"
    on_violation: Literal["warn", "block", "quarantine"] = "warn"
    max_tool_calls_per_turn: int = 20
    max_memory_entries: int = 1000

File: typescript/src/policy.ts
export type ViolationMode = "warn" | "block" | "quarantine";
export type Sensitivity = "low" | "medium" | "high";

export interface Policy {
  toolAllowlist?: string[];
  toolDenylist?: string[];
  memoryTTL?: number; // seconds; 0 = no expiry
  injectionSensitivity?: Sensitivity;
  onViolation?: ViolationMode;
  maxToolCallsPerTurn?: number;
  maxMemoryEntries?: number;
}

export const defaultPolicy = (): Required<Policy> => ({
  toolAllowlist: undefined as unknown as string[],
  toolDenylist: [],
  memoryTTL: 3600,
  injectionSensitivity: "medium",
  onViolation: "warn",
  maxToolCallsPerTurn: 20,
  maxMemoryEntries: 1000,
});
